### PR TITLE
handle more cases for prefer_asserts_in_initializer_list

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: test/rules/prefer_asserts_in_initializer_list.dart [hunk 1]
```diff
@@ -48,13 +51,48 @@ class A {
   // no lint if method is call on other objet
```
Comment (bwilkerson) [inline test/rules/prefer_asserts_in_initializer_list.dart]: "objet" --> "object"

This comment says "no lint", but the lines below are marked with "LINT". Seems inconsistent.

File: test/rules/prefer_asserts_in_initializer_list.dart [hunk 2]
```diff
@@ -48,13 +51,48 @@ class A {
...
+  // lint for call of static member
+  static get sa => null;
+  static sm() => null;
+  A.c13() {
+    assert(sa != null); // LINT
+    assert(sm() != null); // LINT
+  }
+}
+
+// no lint for super class attributs
```
Comment (bwilkerson) [inline test/rules/prefer_asserts_in_initializer_list.dart]: "attributs" --> "attributes"

File: test/rules/prefer_asserts_in_initializer_list.dart [hunk 3]
```diff
@@ -48,13 +51,48 @@ class A {
...
+  get b => null;
+}
+class C extends B {
+  C() {
+    assert(a != null); // OK
+    assert(b != null); // OK
+  }
+}
+
+// no lint for mixin attributs
```
Comment (bwilkerson) [inline test/rules/prefer_asserts_in_initializer_list.dart]: "attributs" --> "attributes"

Comment (a14n) [inline test/rules/prefer_asserts_in_initializer_list.dart]: done

File: test/rules/prefer_asserts_in_initializer_list.dart (hunk 2 above)
Comment (a14n) [inline test/rules/prefer_asserts_in_initializer_list.dart]: done

File: test/rules/prefer_asserts_in_initializer_list.dart (hunk 1 above)
Comment (a14n) [inline test/rules/prefer_asserts_in_initializer_list.dart]: done

**Pull request merged**